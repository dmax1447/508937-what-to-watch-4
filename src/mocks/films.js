
const films = [
  {
    id: `f01`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: `f02`,
    title: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
  },
  {
    id: `f03`,
    title: `Macbeth`,
    picture: `macbeth.jpg`,
  },
  {
    id: `f04`,
    title: `Aviator`,
    picture: `aviator.jpg`,
  },
  {
    id: `f05`,
    title: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: `f06`,
    title: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
  },
  {
    id: `f07`,
    title: `Revenant`,
    picture: `revenant.jpg`,
  },
  {
    id: `f08`,
    title: `Johnny English`,
    picture: `johnny-english.jpg`,
  },
];

export default films;
