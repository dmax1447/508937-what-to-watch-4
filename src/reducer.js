import {extend} from "./utils.js";

const initialState = {
  genre: `All_genres`,
};

const ActionType = {
  SET_GENRE: `SET_GENGE`,
  SET_ALL_GENERES: `SET_ALL_GENERES`,
};

const ActionCreator = {
  setGenre: (genre) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  setAllGeneres: () => ({
    type: ActionType.SET_ALL_GENERES,
    payload: null,
  }),
};

const genres = [
  `Drama`,
  `Adventure`,
  `Bio`,
  `Mystery`,
  `Horror`,
  `Comedy`,
];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case ActionType.SET_ALL_GENERES:
      return extend(state, {
        genre: `All_genres`,
      });
  }

  return state;
};


export {reducer, ActionType, ActionCreator, genres};
