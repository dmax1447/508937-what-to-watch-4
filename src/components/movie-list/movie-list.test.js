import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list.jsx";

import films from '../../mocks/tests/films.js';
import {createStore} from "redux";
import {Provider} from "react-redux";
import {reducer} from "../../reducer.js";

const store = createStore(reducer);

const onCardTitleClick = (evt) => {
  evt.preventDefault();
};

it(`<MovieList /> should render 8 cards`, () => {
  const card = renderer
        .create(
            <Provider store={store}>
              <MovieList films={films} onCardTitleClick={onCardTitleClick} />
            </Provider>
        )
        .toJSON();

  expect(card).toMatchSnapshot();
});
