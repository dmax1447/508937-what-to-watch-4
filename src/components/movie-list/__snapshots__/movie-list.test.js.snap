// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList /> should render 8 cards 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        onClick={[Function]}
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f01"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        onClick={[Function]}
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f02"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        onClick={[Function]}
        src="img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f03"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height="175"
        onClick={[Function]}
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f04"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="We need to talk about Kevin"
        height="175"
        onClick={[Function]}
        src="img/we-need-to-talk-about-kevin.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f05"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="What We Do in the Shadows"
        height="175"
        onClick={[Function]}
        src="img/what-we-do-in-the-shadows.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f06"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Revenant"
        height="175"
        onClick={[Function]}
        src="img/revenant.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f07"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Johnny English"
        height="175"
        onClick={[Function]}
        src="img/johnny-english.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/movie-page/f08"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
